relpipe: init at 0.17.1

MostAwesomeDude opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

MostAwesomeDude committed:
relpipe: init at 0.16

MostAwesomeDude committed:
relpipe: bash completion

MostAwesomeDude commented:
While testing the functionality in `relpipe-in-sql`, I tripped over the issue that is now fixed by #89879. That PR doesn't block this one, but does interfere constructively.

SuperSandro2000 commented on pkgs/tools/text/relpipe/default.nix:
```diff
@@ -0,0 +1,188 @@
+      };
+
+      buildInputs = deps;
+      nativeBuildInputs = [ cmake pkgconfig installShellFiles ];
```
```suggestion
      nativeBuildInputs = [ cmake pkg-config installShellFiles ];
```

SuperSandro2000 commented on pkgs/tools/text/relpipe/default.nix:
```diff
@@ -0,0 +1,188 @@
+{ lib, stdenv, fetchhg, cmake, pkgconfig, installShellFiles, symlinkJoin,
```
```suggestion
{ lib, stdenv, fetchhg, cmake, pkg-config, installShellFiles, symlinkJoin
```

SuperSandro2000 commented on pkgs/tools/text/relpipe/default.nix:
```diff
@@ -0,0 +1,188 @@
+{ lib, stdenv, fetchhg, cmake, pkgconfig, installShellFiles, symlinkJoin,
+  xercesc, libxmlxx, guile_2_2, unixODBC, python36, libjack2 }:
```
```suggestion
, xercesc, libxmlxx, guile_2_2, unixODBC, python36, libjack2 }:

```

SuperSandro2000 commented on pkgs/tools/text/relpipe/default.nix:
```diff
@@ -0,0 +1,188 @@
+        fi
+      '';
+
+      meta = {
+        maintainers = [ stdenv.lib.maintainers.MostAwesomeDude ];
+        homepage = https://relational-pipes.globalcode.info/;
+        license = stdenv.lib.licenses.gpl3;
+        description = "An open data format designed for streaming structured data between two processes";
+      };
```
```suggestion
      meta = with stdenv.lib; {
        description = "An open data format designed for streaming structured data between two processes";
        homepage = "https://relational-pipes.globalcode.info/";
        license = licenses.gpl3;
        maintainers = with maintainers; [ MostAwesomeDude ];
      };
```

SuperSandro2000 commented:
Please remove the merge conflict.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 89787` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>relpipe</li>
  </ul>
</details>

SuperSandro2000 commented:
Please remove the merge master commit.

MostAwesomeDude committed:
relpipe: Address review feedback.

MostAwesomeDude committed:
relpipe: 0.16 -> 0.17.1

Tested a couple of the utilities. I don't have a setup for testing the
JACK tools though.

doronbehar commented on pkgs/tools/text/relpipe/default.nix:
```diff
@@ -0,0 +1,192 @@
+  };
+in symlinkJoin {
+  name = "relpipe";
+  paths = [
```
It would be nice if it was possible to disable some of these packages, to spare some closure size for those who wish so.

MostAwesomeDude replied:
Sure. Or we could split them into an attrset. The upstream intent is that a metapackage will have all of the utilities, but there's nothing wrong with exposing them ala carte.

doronbehar reviewed (commented):
Very good looking package!

MostAwesomeDude committed:
Apply suggestions from code review

And do a couple fixups so that the package still builds.

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>
Co-authored-by: Doron Behar <doron.behar@gmail.com>

MostAwesomeDude commented on pkgs/tools/text/relpipe/default.nix:
```diff
@@ -0,0 +1,192 @@
+      meta = with lib; {
+        maintainers = [ maintainers.MostAwesomeDude ];
+        homepage = https://relational-pipes.globalcode.info/;
+        license = licenses.gpl3;
```
Done.

MostAwesomeDude committed:
relpipe: Use gpl3Plus for license.

This is more accurate. Requested from code review.

MostAwesomeDude commented:
I think that I've handled all review comments, save for the following:
* Closure size: Yes, let's go smaller. I don't know what folks would prefer to do, but I would be okay with putting e.g. `relpipe.in-csv` in there. Do we just make an attrset and merge it into the symlink farm? This seems like a nice feature that doesn't have to block this PR.
* Python version: `relpipe-tr-python` is hardcoded upstream to build against Python 3.6 alone. This can probably be patched later in another PR, if we need more flexibility. I already tried the simple obvious tricks for making the build instructions more flexible, but they did not work.
* Rebase: This branch is relatively old for not having any merge conflicts! I don't think I need to rebase yet though.

Thanks for coming with me on this marathon. I'm renaming the PR to indicate that the merge would make 0.17.1 available, not 0.16.

SuperSandro2000 commented on pkgs/tools/text/relpipe/default.nix:
```diff
@@ -0,0 +1,195 @@
+
+      meta = with lib; {
+        maintainers = [ maintainers.MostAwesomeDude ];
+        homepage = https://relational-pipes.globalcode.info/;
```
```suggestion
        homepage = "https://relational-pipes.globalcode.info/";
```

Lassulus commented:
so, any progress? need any help?

MostAwesomeDude commented:
Yeah, I'll let somebody else finish this. Thanks for reviewing.
